### PR TITLE
Upgrade to Ember Power Select 3

### PR DESCRIPTION
**Pull request by mupkoo**
This PR upgrades Ember to 3.11 and to Ember Power Select 3. 

There are some breaking changes in order to align both addons
1. `onchange` is renamed to `onChange`
2. `oncreate` is renamed to `onCreate`

### CODE REVIEW

File: addon/components/power-select-multiple-with-create.js [hunk 1]
```diff
@@ -5,15 +5,21 @@ export default PowerSelectWithCreate.extend({
   layout,
   powerSelectComponentName: 'power-select-multiple',
 
-  selectOrCreate(selection, select) {
-    let suggestion = selection.filter((option) => {
-      return option.__isSuggestion__;
-    })[0];
+  // Probably needs to be fixed upstream
+  triggerComponent: 'power-select-multiple/trigger',
+  beforeOptionsComponent: null,
```
Comment (mupkoo) [inline addon/components/power-select-multiple-with-create.js]: Without those two lines here, the PowerSelect is rendered with the default components for a single select

File: addon/templates/components/power-select-with-create.hbs [hunk 2]
```diff
@@ -1,64 +1,66 @@
...
+{{#let (component this.powerSelectComponentName) as |Select|}}
+  <Select
+    @afterOptionsComponent={{@afterOptionsComponent}}
+    @allowClear={{@allowClear}}
+    @ariaDescribedBy={{@ariaDescribedBy}}
+    @ariaInvalid={{@ariaInvalid}}
+    @ariaLabel={{@ariaLabel}}
+    @ariaLabelledBy={{@ariaLabelledBy}}
+    @beforeOptionsComponent={{this.beforeOptionsComponent}}
+    {{!-- @buildSelection={{@buildSelection}} --}}
```
Comment (mupkoo) [inline addon/templates/components/power-select-with-create.hbs]: If you allow the user to overwrite `buildSelection` like that `@buildSelection={{@buildSelection}}`, but there is no `buildSelection` passed, the component falls back to `option => option` one, even with `<PowerSelectMultiple>`

**Comment by mupkoo** [comment]
There were couple of strange behaviours as I explained. I am not sure if this is something that needs to be address in the PowerSelect component or in Ember.JS.

File: addon/components/power-select-multiple-with-create.js (hunk 1 above)
Comment (jelhan) [reply addon/components/power-select-multiple-with-create.js]: This should be fixed by https://github.com/cibernox/ember-power-select/pull/1271 upstream.

File: addon/templates/components/power-select-with-create.hbs (hunk 2 above)
Comment (jelhan) [reply addon/templates/components/power-select-with-create.hbs]: This should be fixed by https://github.com/cibernox/ember-power-select/pull/1271 as well.

File: addon/templates/components/power-select-with-create.hbs [hunk 3]
```diff
@@ -1,64 +1,66 @@
...
+    @selected={{@selected}}
+    @selectedItemComponent={{@selectedItemComponent}}
+    @tabindex={{@tabindex}}
+    @triggerClass={{@triggerClass}}
+    @triggerComponent={{this.triggerComponent}}
+    @triggerId={{@triggerId}}
+    @triggerRole={{@triggerRole}}
+    @typeAheadMatcher={{@typeAheadMatcher}}
+    @verticalPosition={{@verticalPosition}}
+    as |option term|
```
Comment (jelhan) [inline addon/templates/components/power-select-with-create.hbs]: Should forward `...attributes`.

File: addon/templates/components/power-select-with-create.hbs (hunk 2 above)
Comment (mupkoo) [reply addon/templates/components/power-select-with-create.hbs]: Looks like it is working after upgrading to 3.0.4

**Comment by mupkoo** [comment]
I updated the PR

File: addon/components/power-select-with-create.js [hunk 4]
```diff
@@ -12,11 +12,12 @@ export default Component.extend({
   matcher: defaultMatcher,
   suggestedOptionComponent: 'power-select-with-create/suggested-option',
   powerSelectComponentName: 'power-select',
+  searchEnabled: true,
```
Comment (jelhan) [inline addon/components/power-select-with-create.js]: This is restoring the old default of `ember-power-select@^2.0.0`, isn't it? Shouldn't `<PowerSelectWithCreate>` have the same defaults as `<PowerSelect>`?

**Comment by cibernox** [comment]
I'm not promising anything, but I'll try to find time this week, probably on Friday, to help make the tests pass if you don't fix them first

**Comment by jelhan** [comment]
@mupkoo If you are struggling with altering ember-try and TravisCI configuration, have a look here: https://github.com/kaliber5/ember-bootstrap-power-select/pull/86 You might be able to just cherry-pick `config/ember-try.js` and `.travis.yml` from that PR.

File: addon/components/power-select-with-create.js (hunk 4 above)
Comment (jelhan) [reply addon/components/power-select-with-create.js]: :+1: That makes sense!

**Comment by mupkoo** [comment]
I will check them a bit later today if I have time.

**Comment by mupkoo** [comment]
The tests are passing 🎉

**Comment by jelhan** [comment]
@cibernox I think this is ready to be merged.

**Comment by jelhan** [comment]
@cibernox Anything I could help you with to get this one merged and released?

**Comment by cibernox** [comment]
Nothing really, I missed this. Thanks!!

**Comment by cibernox** [comment]
Published in 0.7.0

**Pull request merged**